fe/fe-core/src/main/java/com/starrocks/clone/TabletChecker.java

```diff
@@ -902,9 +903,9 @@ private static LocalTabletHealthStats collectLocalTabletHealthStats(LocalTablet
                 stats.incrLocationMatchCnt();
             } else {
                 Pair<String, String> backendLocKV = backend.getSingleLevelLocationKV();
-                if (backendLocKV != null) {
-                    if (!uniqueReplicaLocations.contains(backendLocKV) &&
-                            isLocationMatch(backendLocKV.first, backendLocKV.second, requiredLocation)) {
+                if (backendLocKV != null && isLocationMatch(backendLocKV.first, backendLocKV.second, requiredLocation)) {
```

Comment:
The logic here should be:
If replica HA is not required, just increase the location mach cnt `stats.incrLocationMatchCnt()`
If replica HA should be ensured, increase the location only when uniqueReplicaLocations dose not contains the be's backendLockKV.
So the code should be:
```
if (ensureReplicaHA) {
    stats.incrLocationMatchCnt();
else if (!uniqueReplicaLocations.contains(backendLocKV)) {
    stats.incrLocationMatchCnt();
    uniqueReplicaLocations.add(backendLocKV);
}

```

Reply:
You code seems works well, but we should make the logic more clear.